components/ui-api-layer/internal/domain/remoteenvironment/re_service.go

```diff
@@ -67,6 +69,47 @@ func newRemoteEnvironmentService(client remoteenvironmentv1alpha1.Applicationcon
 	}, nil
 }
 
+func (svc *remoteEnvironmentService) Create(name string, description string, labels map[string]interface{}) (*v1alpha1.RemoteEnvironment, error) {
+	return svc.client.RemoteEnvironments().Create(&v1alpha1.RemoteEnvironment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RemoteEnvironment",
+			APIVersion: "applicationconnector.kyma-project.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.RemoteEnvironmentSpec{
+			Labels:      svc.convertJsonToLabels(labels),
+			Description: description,
+			Services:    []v1alpha1.Service{},
+		},
+	})
+}
+
+func (svc *remoteEnvironmentService) Update(name string, description string, labels map[string]interface{}) (*v1alpha1.RemoteEnvironment, error) {
```

Comment:
could you rethink the idea of the labels arg?

it looks like this:
*root_resolver.go:* UpdateRemoteEnvironment method has labels ***gqlschema.JSON** -> *re_resolver.go:*  UpdateRemoteEnvironment method has labels ***gqlschema.JSON** -> *re_service.go:* Update method has **map[string]interface{}** and then you are converting it to **map[string]string**

so use the ***gqlschema.JSON** in Update method in service or just convert it in resolver to **map[string]string**